app/src/main/java/org/simple/clinic/widgets/PrimarySolidButtonWithFrame.kt

```diff
@@ -0,0 +1,66 @@
+package org.simple.clinic.widgets
+
+import android.content.Context
+import android.support.annotation.ColorInt
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import org.simple.clinic.R
+
+class PrimarySolidButtonWithFrame(context: Context, attributeSet: AttributeSet?) : FrameLayout(context, attributeSet) {
+
+  val button by lazy { getChildAt(0) as PrimarySolidButton }
+
+  @ColorInt
+  private var frameBackgroundEnabledResId: Int = 0
+
+  @ColorInt
+  private var frameBackgroundDisabledResId: Int = 0
+
+  @ColorInt
+  private var buttonBackgroundEnabledResId: Int = 0
+
+  @ColorInt
+  private var buttonBackgroundDisabledResId: Int = 0
```

Comment:
This was added because I explicitly want to verify that all attributes have been set in the XML. Setting it to `Color.BLACK` makes it non-zero and unverifiable.